Open Judge System/Data/OJS.Data/Repositories/SubmissionsForProcessingRepository.cs

```diff
@@ -10,5 +14,71 @@ public SubmissionsForProcessingRepository(IOjsDbContext context)
             : base(context)
         {
         }
+
+        public void AddOrUpdate(int submissionId)
+        {
+            var submissionForProcessing = this.Context.SubmissionsForProcessing
+                .FirstOrDefault(sfp => sfp.SubmissionId == submissionId);
+
+            if (submissionForProcessing != null)
+            {
+                submissionForProcessing.Processing = false;
+                submissionForProcessing.Processed = false;
+            }
+            else
+            {
+                submissionForProcessing = new SubmissionForProcessing
+                {
+                    SubmissionId = submissionId
+                };
+
+                this.Context.SubmissionsForProcessing.Add(submissionForProcessing);
+            }
+        }
+
+        public void AddOrUpdate(IEnumerable<int> submissionIds)
+        {
+            try
+            {
+                this.Context.DbContext.Configuration.AutoDetectChangesEnabled = false;
+                foreach (var submissionId in submissionIds)
+                {
+                    var submissionForProcessing = this.Context.SubmissionsForProcessing
```

Comment:
Unify logic between this method and the single AddOrUpdate and reuse common functionality.